pkg/controller/studyjob/katib_api_util.go

```diff
@@ -268,6 +277,52 @@ func createStudy(c katibapi.ManagerClient, studyConfig *katibapi.StudyConfig) (s
 	return studyID, nil
 }
 
+func registerStudyJob(c katibapi.ManagerClient, instance *katibv1alpha1.StudyJob) error {
+	ctx := context.Background()
+	regStudyJobreq := &katibapi.RegisterStudyJobRequest{
+		StudyJob: &katibapi.StudyJob{
+			StudyJobUid: string(instance.UID),
+			StudyId:     instance.Status.StudyID,
+		},
+	}
+	if instance.Spec.WorkerSpec != nil {
+		if instance.Spec.WorkerSpec.GoTemplate.RawTemplate != "" {
+			regStudyJobreq.StudyJob.WorkerTemplate = instance.Spec.WorkerSpec.GoTemplate.RawTemplate
+		} else {
+			sjc, err := studyjobclient.NewStudyjobClient(nil)
+			if err != nil {
+				return err
+			}
+			wtl, err := sjc.GetWorkerTemplates()
+			if err != nil {
+				return err
+			}
+			if wt, ok := wtl[instance.Spec.WorkerSpec.GoTemplate.TemplatePath]; ok {
+				regStudyJobreq.StudyJob.WorkerTemplate = wt
+			}
+		}
+	}
+	if instance.Spec.MetricsCollectorSpec != nil {
+		if instance.Spec.MetricsCollectorSpec.GoTemplate.RawTemplate != "" {
+			regStudyJobreq.StudyJob.MetricsCollectorTemplate = instance.Spec.MetricsCollectorSpec.GoTemplate.RawTemplate
+		} else {
+			sjc, err := studyjobclient.NewStudyjobClient(nil)
+			if err != nil {
+				return err
+			}
+			mtl, err := sjc.GetMetricsCollectorTemplates()
+			if err != nil {
+				return err
+			}
+			if mt, ok := mtl[instance.Spec.MetricsCollectorSpec.GoTemplate.TemplatePath]; ok {
+				regStudyJobreq.StudyJob.MetricsCollectorTemplate = mt
+			}
+		}
```

Comment:
please also handle `GoTemplate.TemplatePath == "" && GoTemplate.RawTemplate == "" ` case both for worker and metrics.

Reply:
In that case, the `defaultWorkerTemplate.yaml` and `defaultMetricsCollectorTemplate.yaml` will be used. It's not an error.